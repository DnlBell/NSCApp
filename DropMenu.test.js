import React from 'react';
import MenuItem from './components/DropMenu';
import renderer from 'react-test-renderer';

const options = ["Home", "Login", "Register"];

describe('dropdown menu', ()=> {
    it('renders an empty button initially', () => {
        const rendered = renderer.create(<MenuItem />).toJSON();
        expect(rendered).toMatchSnapshot();
      });

    it('renders a list of options on press', () => {
        const rendered = renderer.create(<MenuItem />).toJSON();
        expect(rendered).toBeTruthy();
      });
})