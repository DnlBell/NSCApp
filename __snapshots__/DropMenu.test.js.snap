// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dropdown menu renders an empty button initially 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "paddingLeft": 15,
      "paddingTop": 30,
    }
  }
>
  <View
    collapsable={false}
  >
    <View
      onLayout={[Function]}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        onPress={[Function]}
      >
        Mindspand
      </Text>
    </View>
    <Modal
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      supportedOrientations={
        Array [
          "portrait",
          "portrait-upside-down",
          "landscape",
          "landscape-left",
          "landscape-right",
        ]
      }
      transparent={true}
      visible={false}
    >
      <View
        accessible={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          onLayout={[Function]}
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 4,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.14,
              "shadowRadius": 2,
              "top": 0,
              "transform": Array [],
            }
          }
        >
          <View
            style={
              Object {
                "overflow": "hidden",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "height": 48,
                    "justifyContent": "center",
                    "maxWidth": 248,
                    "minWidth": 124,
                  },
                  undefined,
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="clip"
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "fontSize": 14,
                      "fontWeight": "400",
                      "paddingHorizontal": 16,
                    },
                    false,
                    undefined,
                  ]
                }
              >
                Home
              </Text>
            </View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "height": 48,
                    "justifyContent": "center",
                    "maxWidth": 248,
                    "minWidth": 124,
                  },
                  undefined,
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="clip"
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "fontSize": 14,
                      "fontWeight": "400",
                      "paddingHorizontal": 16,
                    },
                    false,
                    undefined,
                  ]
                }
              >
                Login
              </Text>
            </View>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "height": 48,
                    "justifyContent": "center",
                    "maxWidth": 248,
                    "minWidth": 124,
                  },
                  undefined,
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="clip"
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "fontSize": 14,
                      "fontWeight": "400",
                      "paddingHorizontal": 16,
                    },
                    false,
                    undefined,
                  ]
                }
              >
                Register
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "borderBottomWidth": 0.5,
                    "flex": 1,
                  },
                  Object {
                    "borderBottomColor": "rgba(0,0,0,0.12)",
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
    </Modal>
  </View>
</View>
`;
